src/ws/cockpitauth.c

```diff
@@ -1502,11 +1502,20 @@ cockpit_auth_login_async (CockpitAuth *self,
 
   /* If the client sends a TLS certificate to cockpit-tls, treat this as a
    * definitive login type, and don't just silently fall back to other types */
-  const gchar *client_certificate;
   JsonObject *metadata = get_connection_metadata (connection);
-  if (metadata && (client_certificate = json_object_get_string_member (metadata, "client-certificate")))
+  if (metadata && json_object_has_member (metadata, "client-certificate"))
```

Comment:
I feel like this code is trying to basically duplicate the cockpit json helpers here...

Reply:
Right, this could use cockpit_json_get_string().

Reply:
Done.